### PR TITLE
[POAE7-2752] Enable cpplint of cider/common

### PR DESCRIPTION
**Pull request by Deegue**
### What changes were proposed in this pull request?
Enable cpplint of cider/common.

### Why are the changes needed?
Improvement.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
CI

### CODE REVIEW

File: src/cider/common/hashtable/FixedHashMap.h [hunk 1]
```diff
@@ -36,12 +36,12 @@ struct FixedHashMapCell {
   bool full;
   Mapped mapped;
 
-  FixedHashMapCell() {}  //-V730 /// NOLINT
+  FixedHashMapCell() {}  // NOLINT
```
Comment (jikunshang) [inline src/cider/common/hashtable/FixedHashMap.h:39]: seems can remove `// NOLINT` ?

File: src/cider/common/hashtable/HashTable.h [hunk 2]
```diff
@@ -1077,15 +1071,11 @@ class HashTable : private boost::noncopyable,
     return const_cast<std::decay_t<decltype(*this)>*>(this)->find(x, hash_value);
   }
 
-  ALWAYS_INLINE bool erase(const Key& x)
   // requires Grower::performs_linear_probing_with_single_step
-  {
-    return erase(x, hash(x));
-  }
+  ALWAYS_INLINE bool erase(const Key& x) { return erase(x, hash(x)); }
```
Comment (jikunshang) [inline src/cider/common/hashtable/HashTable.h:1075]: where defined `ALWAYS_INLINE`

File: src/cider/common/hashtable/FixedHashMap.h (hunk 1 above)
Comment (Deegue) [reply src/cider/common/hashtable/FixedHashMap.h]: removed

File: src/cider/common/hashtable/HashTable.h (hunk 2 above)
Comment (Deegue) [reply src/cider/common/hashtable/HashTable.h:1074]: It seems to put `ALWAYS_INLINE` behind return type info in velox and ck.

Comment (jikunshang) [reply src/cider/common/hashtable/HashTable.h:1074]: `ALWAYS_INLINE` is defined in `funcannotations.h`, files in common folder will include this header file?

**Review by jikunshang — APPROVED** [review]

File: src/cider/common/hashtable/HashTable.h (hunk 2 above)
Comment (Deegue) [reply src/cider/common/hashtable/HashTable.h:1074]: I got it, I think it should be included but it's strange that it compiled well now..

**Review by kd97 — APPROVED** [review]

**Pull request merged**